github/with_appengine.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// This file provides glue for making github work on App Engine.
+
+// +build appengine
+
+package github
+
+import (
+	"context"
```

Comment:
Although now that I'm thinking about this a little more, I'm wondering how the other `import "context"` in the rest of this package are going to work.

I think I'm going to have to do some more experimentation with this PR before we call it ready for review.  Sorry, @shurcooL.